src/main/java/com/ethercamp/harmony/web/controller/ContractsController.java

```diff
@@ -40,87 +44,106 @@
-    public boolean stopWatchingContract(@PathVariable String address) {
+    @RequestMapping(value = "/{address}/delete", method = RequestMethod.POST)
+    public boolean stopWatchingContract(@PathVariable String address) throws DisabledException {
         return contractsService.deleteContract(address);
     }
 
-    @RequestMapping(value = "/contracts/{address}/files", method = RequestMethod.POST)
+    @RequestMapping(value = "/{address}/files", method = RequestMethod.POST)
     public ActionStatus<ContractInfoDTO> uploadContractFiles(
             @PathVariable String address,
             @RequestParam MultipartFile[] contracts,
-            @RequestParam(required = false) String verifyRlp) {
+            @RequestParam(required = false) String verifyRlp) throws DisabledException {
 
         try {
             ContractInfoDTO contract = contractsService.uploadContract(lowerCase(address), contracts);
             log.info("Uploaded files for address: {}, contract name: {}" + address, contract.getName());
             return createSuccessStatus(contract);
+        } catch (DisabledException e) {
+            throw e;
         } catch (Exception e) {
             log.warn("Contract's source uploading error: ", e);
             return createErrorStatus(e.getMessage());
         }
     }
 
-    @RequestMapping(value = "/contracts/{address}/importFromExplorer", method = RequestMethod.POST)
-    public ActionStatus<Boolean> importContractDataFromExplorer(@PathVariable String address) {
+    @RequestMapping(value = "/{address}/importFromExplorer", method = RequestMethod.POST)
+    public ActionStatus<Boolean> importContractDataFromExplorer(@PathVariable String address) throws DisabledException {
         try {
             final boolean result = contractsService.importContractFromExplorer(address);
             return createSuccessStatus(result);
+        } catch (DisabledException e) {
+            throw e;
         } catch (Exception e) {
             return createErrorStatus(e.getMessage());
         }
     }
 
-    @RequestMapping(value = "/contracts/{address}/clearContractStorage", method = RequestMethod.POST)
-    public ActionStatus<Boolean> clearStorage(@PathVariable String address) {
+    @RequestMapping(value = "/{address}/clearContractStorage", method = RequestMethod.POST)
+    public ActionStatus<Boolean> clearStorage(@PathVariable String address) throws DisabledException {
         try {
             contractsService.clearContractStorage(address);
             return createSuccessStatus();
+        } catch (DisabledException e) {
+            throw e;
         } catch (Exception e) {
             return createErrorStatus(e.getMessage());
         }
     }
 
-    @RequestMapping("/contracts/indexStatus")
-    public ActionStatus<IndexStatusDTO> getIndexStatus() {
+    @RequestMapping("/indexStatus")
+    public ActionStatus<IndexStatusDTO> getIndexStatus() throws DisabledException {
```

Comment:
`DisabledException` looks like a `RuntimeException`. is there any reason for it being defined in another way?

Reply:
Hmmm. ok, we could remove "throws" in method description going this way